vm.go

```diff
@@ -536,79 +556,174 @@ func (v *VM) run() {
 				v.sp++
 			}
 		case parser.OpCall:
-			numArgs := int(v.curInsts[v.ip+1])
-			spread := int(v.curInsts[v.ip+2])
-			v.ip += 2
+			var (
+				numArgs           = int(v.curInsts[v.ip+1])
+				hasVarArgs        = int(v.curInsts[v.ip+2])
+				numKws            = int(v.curInsts[v.ip+3])
+				hasVarKw          = int(v.curInsts[v.ip+4])
+				args              []Object
+				kwargs, varKwargs map[string]Object
+				hasKws            int
+			)
+
+			v.ip += 4
+			if numKws > 0 {
+				hasKws = 1
+			}
+
+			start := v.sp - numArgs - hasVarArgs - hasKws - hasVarKw
+			value := v.stack[start-1]
 
-			value := v.stack[v.sp-1-numArgs]
 			if !value.CanCall() {
 				v.err = fmt.Errorf("not callable: %s", value.TypeName())
 				return
 			}
 
-			if spread == 1 {
-				v.sp--
-				switch arr := v.stack[v.sp].(type) {
+			for i := 0; i < numArgs; i++ {
+				args = append(args, v.stack[start+i])
```

Comment:
This is probably playing a major part. Don't allocate new memory if you can work with existing. If you absolutely need this one create it with the size you need.